### PR TITLE
Create fixture for spawner specific configurations

### PR DESCRIPTION
**Pull request by Sheila-nk**
- created a fixture that allows tests to configure attributes specific to the spawner implementation

Reference #20

### CODE REVIEW

**Comment by Sheila-nk** [comment]
@GeorgianaElena 👋 let me know if this PR works for task 3 of issue #20

**Review by GeorgianaElena — CHANGES_REQUESTED** [review]
Thank you for working on this one @Sheila-nk! 

> @GeorgianaElena 👋 let me know if this PR works for task 3 of issue https://github.com/jupyterhub/pytest-jupyterhub/issues/20

Please checkout https://github.com/jupyterhub/pytest-jupyterhub/issues/25#issuecomment-1403453652 and the top comment where I tried to put everything into context, and let me know if it clarifies a bit more the purpose of this fixture. But please feel free to ask about the bits that are still confusing.

**Comment by Sheila-nk** [comment]
This part from issue #25 confuses me the most:

> focus on finding ways to make anything that seems specific to a spawner implementation be configurable by a spawner in its own test suite

You had mentioned that this fixture should return an empty config. Does that look like this:
```python
@pytest.fixture
def spawner_config():
    """Allows tests to setup configurations that are specific to the spawner implementation"""
    spawner_config = {}
    return Config(spawner_config)
```
We just leave the spawner test suite to specify its attributes within this fixture, then merge the resulting "dict" with the `jupyterhub_spawner_class_config` "dict" and use that config to create an instance of the MockHub?

**Comment by GeorgianaElena** [comment]
> This part from issue https://github.com/jupyterhub/pytest-jupyterhub/issues/25 confuses me the most:

>  > focus on finding ways to make anything that seems specific to a spawner implementation be configurable by a spawner in its own test suite
    
I'm sorry this is confusing. Let me try and give more insight about it:

Because there is no way of covering all the configuration setups needed by a Spawner implementation in its test suite, and because we want this pytest plugin to be **useful** for future spawners also, that don't exist yet, there is no way to know how to configure the hub app for such tests from within the pytest plugin repo, right?

But what **do know** is that almost if not all spawners configure the hub app in some way before starting it and before running their test against it.

The hub app fixture that we are currently trying to implement should be responsible for executing the set of steps that are common for all the spawners, in the correct order, but at the same time, leave the steps that are specific to an implementation be configurable.

So what we want, is for _all_ the spawners to be able to use the same, **existing** fixture that's implemented and maintained in this pytest plugin repo with proper specific configuration.

> You had mentioned that this fixture should return an empty config.

Because the fixture in this plugin doesn't know and doesn't need to know every spawner internal config or testing intentions, we should leave it empty and use it as a placeholder in the `app_for_spawners` fixture.

**Comment by Sheila-nk** [comment]
@GeorgianaElena Thank you for the in-depth explanation. I think I am getting it now.

File: pytest_jupyterhub/jupyterhub_spawners.py [hunk 1]
```diff
@@ -19,3 +19,10 @@ def jupyterhub_spawner_class_config(spawner_class):
     """Configures JupyterHub to use the Spawner class specified in the spawner_class fixture."""
     spawner_class_config = {"JupyterHub": {"spawner_class": spawner_class}}
     return Config(spawner_class_config)
+
+
+@pytest.fixture
+def spawner_config():
+    """Allows tests to setup configurations that are specific to the spawner implementation"""
+    spawner_config = {}
+    return Config(spawner_config)
```
Comment (GeorgianaElena) [inline pytest_jupyterhub/jupyterhub_spawners.py:28]:
```suggestion
    return Config()
```

Comment (GeorgianaElena) [reply pytest_jupyterhub/jupyterhub_spawners.py:28]: return Config({})

**Review by GeorgianaElena — APPROVED** [review]
🎉 This is great! Thank you! 🎉
